wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextDsl.kt

```diff
@@ -0,0 +1,421 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qkl.wrapper.minecraft.text
+
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.server.command.ServerCommandSource
+import net.minecraft.text.*
+import net.minecraft.text.data.TextData
+import net.minecraft.util.Identifier
+import org.quiltmc.qkl.wrapper.minecraft.*
+import org.quiltmc.qkl.wrapper.minecraft.isBoldRaw
+import org.quiltmc.qkl.wrapper.minecraft.isItalicRaw
+import org.quiltmc.qkl.wrapper.minecraft.isStrikethroughRaw
+import org.quiltmc.qkl.wrapper.minecraft.isUnderlinedRaw
+import java.util.*
+import java.util.stream.Stream
+
+/**
+ * Marks objects as being part of QKL's Text Builder DSL.
+ */
+@DslMarker
+public annotation class TextDsl
+
+/**
+ * A builder for a styled [Text] instance. Can be built into text with [build].
+ *
+ * @see [buildText]
+ *
+ * @author Cypher121
+ */
+@TextDsl
+public class TextBuilder {
+    /**
+     * Current text state of the builder.
+     */
+    public val text: MutableText = Text.empty()
+
+    /**
+     * Current style state of the builder to be applied with [styleAndAppend].
+     */
+    public val style: StyleBuilder = StyleBuilder()
+
+    /**
+     * Applies the given [action] while setting the [StyleBuilder] property
+     * described by [getProp] and [setProp] to [newValue] for the duration of the action.
+     *
+     * The value is then reset to its original state.
+     *
+     * @author Cypher121
+     */
+    public inline fun <T> withProp(
+        newValue: T,
+        getProp: StyleBuilder.() -> T,
+        setProp: StyleBuilder.(T) -> Unit,
+        action: TextBuilder.() -> Unit
+    ) {
+        val oldValue = getProp(style)
+        setProp(style, newValue)
+
+        action()
+
+        setProp(style, oldValue)
+    }
+
+    /**
+     * Applies the current [style] of the builder to the given [text]
+     * and appends it to the builder's result.
+     */
+    @TextDsl
+    public fun styleAndAppend(text: MutableText) {
+        this.text.append(style.applyTo(text))
+    }
+
+    /**
+     * Returns the [Text] result of this builder.
+     */
+    public fun build(): Text {
+        return text
+    }
+}
+
+/**
+ * Builds a [Text] instance by configuring a [TextBuilder] with the given [action].
+ */
+@TextDsl
+public inline fun buildText(action: TextBuilder.() -> Unit): Text {
+    return TextBuilder().apply(action).build()
+}
+
+/**
+ * Adds a translatable text.
+ *
+ * @param value The translation key of the text.
+ * @param args Any arguments to apply to the translatable text. Can be left
+ * empty for no arguments.
+ * @see StyleBuilder for action
+ *
+ * @author NoComment1105
+ */
+@TextDsl
+public fun TextBuilder.translatable(
+    value: String,
```

Comment:
Surely this and other functions don't exceed the line limit? Could they be put on the same line?

Reply:
Line limit?

Reply:
I'm leaving the NBT ones multiline cause they have a lot of arguments. Everything else moved to single-line

Reply:
All good